plutus-tx/src/PlutusTx/Lift.hs

```diff
@@ -102,7 +102,7 @@ safeLiftCode
        , PLC.Typecheckable uni fun
        , PrettyPrintable uni fun
        )
-    => a -> m (CompiledCodeIn uni fun a)
+    => a -> m (CompiledCodeIn uni fun () a)
```

Comment:
I'm not convinced that every annotation type should have a default. I can perhaps rename `annMayInline` to `annDefault`.

Reply:
Right, I don't love it either. But it's a bit unclear what to do with e.g. `Lift` when it needs a default annotation. There is going to be a lot of churn when we mess with it to add source spans (or not). Sometimes people abuse `Monoid` for that but that's also somewhat wrong. I would also be fine with passing it around in a Reader or otherwise.

Reply:
> There is going to be a lot of churn when we mess with it to add source spans (or not).

I've actually done that in a branch, the change that adds source spans doesn't have a lot to do with annotations.

Reply:
Happy to leave it for now, just a thought!